api-gateway/src/app.ts

```diff
@@ -0,0 +1,54 @@
+import express from 'express';
+import cors from 'cors';
+import routes from './routes';
+import { matchingServiceProxy } from './routes/matchingRoutes';
+import { errorHandler } from './middleware/errorHandler';
+import config from './config';
+import logger from './utils/logger';
+import http from 'http';
+import { Socket } from 'net';
+
+const app = express();
+
+// Middleware
+app.use(cors(config.corsOptions));
+app.use(express.json());
+
+// Logging middleware
+app.use((req, res, next) => {
+  logger.info(`Incoming request: ${req.method} ${req.originalUrl}`);
+  next();
+});
+
+// Routes
+app.use('/api', routes);
```

Comment:
can we add v1 to this? just looks nicer
```suggestion
app.use('/api/v1', routes);
```